### PR TITLE
MOB-1936 Silence precondition with warning on Deferred.swift

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-1936](https://ecosia.atlassian.net/browse/MOB-1936)

## Context

Found multiple crashes on the Deferred object.
The goal is to leverage a possible silencing of the precondition and raise a simple Warning message 

## Approach

Silencing the precondition and replacing it with a warning.

[MOB-1936]: https://ecosia.atlassian.net/browse/MOB-1936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ThirdParty/Deferred/Deferred/Deferred.swift
Comment (lucaschifino) [inline ThirdParty/Deferred/Deferred/Deferred.swift:34]: Might make sense to very briefly mention why here, something like `Ecosia: Commenting since the precondition caused a crash when stored Favicons were "too large"` (not sure if that was actually the case, you can phrase it better 🙈 )

**Review by lucaschifino — COMMENTED** [review]
No blockers from my side, just added a small commenting suggestion.

I guess we want to hold this one until we get an ok from the user somehow? 🤔

File: ThirdParty/Deferred/Deferred/Deferred.swift
Comment (d4r1091) [reply ThirdParty/Deferred/Deferred/Deferred.swift:34]: Will add a better comment 💪 - thanks!

**Review by ecotopian — APPROVED** [review]
LGTM 🟢

File: ThirdParty/Deferred/Deferred/Deferred.swift
Comment (ecotopian) [inline ThirdParty/Deferred/Deferred/Deferred.swift:39]: fyi: formatting seems to be off

Comment (d4r1091) [reply ThirdParty/Deferred/Deferred/Deferred.swift:39]: I wanted to comment purely on that line with the auto-comment `CMD+/` that's why it seems to be off.

**Pull request merged**
